### PR TITLE
Finished claims widget

### PR DESCRIPTION
**Pull request by Tsearo**
Changes out the New Achievements component with Finished Set Claims component

Old
image

New
image

Reasoning being that a majority of the time New Achievements essentially shows that a single game got a set created. Changing to show newly finished sets/revisions show more useful details and if you want to see specific new achievements you will likely be checking the Achievement List page anyways which shows them all.

Finished Claims "more..." link will take you to the Claim List page filtered on Complete statuses.

### CODE REVIEW

File: lib/render/set-claim.php [hunk 1]
```diff
@@ -1,22 +1,32 @@
...
 use RA\ClaimFilters;
+use RA\ClaimSetType;
 use RA\ClaimSorting;
 
 /**
- * Creates the New Set Claims component.
+ * Creates the Set Claims component.
  */
-function renderNewClaimsComponent(int $count): void
+function renderClaimsComponent(int $count, int $claimFilter = ClaimFilters::Default, int $claimSort = ClaimSorting::ClaimDateDescending): void
```
Comment (Jamiras) [inline lib/render/set-claim.php:10]: I'm really struggling with this change. About half of the function is shared, and the other half is conditionally A or B based on a parameter that can have values other than A or B.

I really think this should probably be two separate functions despite the fact that's there's a decent amount of redundancy between them.

**Review by Jamiras — COMMENTED** [review]
Great idea. Seems to work as expected. Left some comments about the implementation.

File: src/ClaimFilters.php [hunk 2]
```diff
@@ -33,6 +33,11 @@ abstract class ClaimFilters
     // This should be updated every time a new filter is added so it has all possible filter bits set
     public const AllFilters = (1 << 13) - 1;
 
+    // Filter to show all Complete Primary claims
+    public const CompletedFilter = self::AllFilters & ~self::ActiveClaim
+                                            & ~self::DroppedClaim
+                                            & ~self::CollaborationClaim;
```
[the remark this answers is not in the document]
Comment (Tsearo) [reply src/ClaimFilters.php:39]: Thanks that is a much better name for this I agree

File: lib/render/set-claim.php (hunk 1 above)
Comment (Tsearo) [reply lib/render/set-claim.php]: I was a bit on the fence about it as well but figured less code was better. I'll split it out though as it's easier to read that way

**Comment by televandalist** [comment]
I find this very enjoyable. Yes.

**Comment by snowpin** [comment]
Ah yea! This makes much more sense to have over a list of achievements

**Review by Jamiras — APPROVED** [review]

File: lib/render/set-claim.php [hunk 3]
```diff
@@ -36,3 +37,40 @@ function renderNewClaimsComponent(int $count): void
...
     echo "</div>";
 }
+
+/**
+ * Creates the Completed Set Claims component.
+ */
+function renderFinishedClaimsComponent(int $count): void
+{
+    echo "<div class='component'>";
+    echo "<h3>Finished Set Claims</h3>";
```
Comment (luchaos) [inline lib/render/set-claim.php:47]: "New Sets"?

File: lib/render/set-claim.php [hunk 4]
```diff
@@ -36,3 +37,40 @@ function renderNewClaimsComponent(int $count): void
...
+
+    $claimData = getFilteredClaimData(0, ClaimFilters::AllCompletedPrimaryClaims, ClaimSorting::FinishedDateDescending, false, null, false, 0, $count);
+
+    echo "<tbody><table>";
+    echo "<th>User</th>";
+    echo "<th>Game</th>";
+    echo "<th nowrap>Type</th>";
+    echo "<th nowrap>Finished On</th>";
```
Comment (luchaos) [inline lib/render/set-claim.php:55]: "On" ca be omitted imo.

**Review by luchaos — APPROVED** [review]

File: lib/render/set-claim.php (hunk 3 above)
Comment (Jamiras) [reply lib/render/set-claim.php:47]: It's also completed revisions, and it complements the "new set claims" that's further down the page.

Comment (luchaos) [reply lib/render/set-claim.php:47]: Was about to write that into a discussion - probably still should. Maybe that's all just a matter of taste.

I find the increasing use of "claims" on the front page problematic.

Claims - while unfortunately necessary - just describe the underlying processes that we had to introduce due to the shortcomings of what the site can do.

Personally, I'd rather focus on what this should become - a more collaborative system. "Claiming" something, to me, implies ownership and that should not have to exist.
Coming to the site as a player "claims" requires more explanation than calling it "new/revised sets" and/or "sets in progress".

Anyways, this probably not the best place to discuss this but wanted to have written down somewhere.

Comment (Tsearo) [reply lib/render/set-claim.php:47]: Would something like `Recently Finished Sets / Revisions` work better in your opinion?

Comment (Jamiras) [reply lib/render/set-claim.php:47]: "Recently Finished" seems overly wordy. Just "New Sets/Revisions" seems fine.

I'm not sure about better wording for "New Set Claims" though. "Sets in Progress" (as suggested above) seems like the best candidate. Then you'd also want to change "Claimed On" to "Started".

Comment (Jamiras) [reply lib/render/set-claim.php]: > Was about to write that into a discussion - probably still should.

This probably deserves a discussion and a separate PR as the "claim" nomenclature permeates the site.

At a minimum, the menu item "Community > Claim List" should also be updated to "Sets in Progress". The rest of the references are arguably acceptable (even the "Claim List" page itself) as it does represent a view of the process.

Comment (luchaos) [reply lib/render/set-claim.php]: I'd see "recently" as implied/superfluous, too. It's the front page so it should be the new stuff only anyways.

Agreed, from a developer's perspective it's all about claims; I'd keep public pages (except for where it's explicitly about claims) targeted at players and go for "sets" as that's what a player interacts with eventually, not underlying claim.

**Pull request merged**